tools/2023-12-test-organization.md

```diff
@@ -0,0 +1,114 @@
+# Organization of functional tests
+
+* **Status**: Pending
+* **Author**: Vinaya Damle (@vinayada1)
+
+## Overview
+
+Currently, all the functional tests require an approval to run which is unnecessary. Only, the tests that use cloud resources need to be approved to run. Also, in the future, we could have tests that require running in AKS/EKS cluster, use a mix of AWS and Azure resources. Currently, we do not have a good plan for organizing such tests.
+
+This is a proposal for organization of functional tests such that it accounts for these requirements and logically organizes the test cases
+
+## Terms and definitions
+
+NA
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6588
+
+### Goals
+
+1. Eliminate the need of an approval for running tests that do not use cloud resources
+2. Account for future tests that might require running in AKS/EKS cluster, a mixed use of AWS and Azure resources, etc.
+
+### Non goals
+
+NA
+
+
+## Design
+
+The proposal is to have the following directory structure:-
+```
+- test
+    - functional-portable (run on any k8s clusters)
+        - ucp
+            - cloud
+            - noncloud		
+                - use features and a single folder for aws and azure
+        - messagingrp
+            - cloud
+            - noncloud		
+                - use features and a single folder for aws and azure
+        - kubernetes
+            - cloud
+            - noncloud		
+                - use features and a single folder for aws and azure
+        - daprrp
+            - cloud
+            - noncloud		
+                - use features and a single folder for aws and azure
+        - datastoresrp
+            - cloud
+            - noncloud		
+                - use features and a single folder for aws and azure
+        - corerp
+            - cloud
+            - noncloud		
+                - use features and a single folder for aws and azure
+        - cli
+            - cloud
+            - noncloud		
+                - use features and a single folder for aws and azure
```

Comment:
Before we merge this plan, can you rearrange this structure to be clearer? This make it look like `noncloud` tests will be testing aws and azure. It should be the cloud tests.